addons/sale_purchase/models/sale_order.py

```diff
@@ -288,7 +288,7 @@ def _purchase_service_create(self, quantity=False):
                 ], limit=1)
             if not purchase_order:
                 values = line._purchase_service_prepare_order_values(supplierinfo)
-                purchase_order = PurchaseOrder.create(values)
+                purchase_order = PurchaseOrder.with_context({'mail_create_nosubscribe': True}).create(values)
```

Comment:
```suggestion
                purchase_order = PurchaseOrder.with_context(mail_create_nosubscribe=True).create(values)
```
`with_context(dict)` replaces the whole context with the given dict.  That's not what we want here, we only want to set/update the `mail_create_nosubscribe´ value.